server/backend/cron/news.go

```diff
@@ -131,53 +135,90 @@ func (c *CronService) parseNewsFeed(source model.NewsSource) error {
 	return nil
 }
 
-// downloadAndSaveImage
-// Downloads an image from the url, converts it to 720p width to and saves it to the database.
-// Returns id of file in db, null int on error.
-func (c *CronService) downloadAndSaveImage(url string) null.Int {
+// getDatabaseIdForImageAndDownload
+// Returns a file id that was saved to the database if it already exists. Otherwise creates it and triggers download of the image
+func (c *CronService) getDatabaseIdForImageAndDownload(url string) (null.Int, error) {
+	targetFileName := fmt.Sprintf("%x.jpg", md5.Sum([]byte(url)))
+	var fileId null.Int
+	if err := c.db.Model(model.Files{}).Where("name = ?", targetFileName).Select("file").Scan(&fileId).Error; err != nil && err != gorm.ErrRecordNotFound {
```

Comment:
This line is super hard to read. If we'd have the query build on the line before, then it might be easier to understand. IDK, up to you.